### PR TITLE
refactor: use `days_early_for_beta` from `InheritanceMixin` instead of runtime [FC-0026]

### PR DESCRIPTION
**Pull request by Agrendalath**
**TL;DR -** Use the block's property directly instead of getting it from the runtime.

**What changed?**

FC-0026 removes the block-specific handling from the runtime. In https://github.com/openedx/edx-platform/pull/32356, we are simplifying user checks. `prepare_runtime_for_user` adds the `days_early_for_beta` attribute to the runtime, but this value is already provided by the `InheritanceMixin`. We are removing this redundancy, so this makes `ora2` retrieve the value directly from XBlock fields.

This also updates the PR template and the release process doc, as they were slightly outdated.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled: n/a
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

1. Go to the [Course Advanced Settings](http://localhost:18010/settings/advanced/course-v1:edX+DemoX+Demo_Course) and set `Days Early for Beta Users` to `6000`.
2. Create an ORA2 XBlock. Go to `Edit -> SCHEDULE` and set its start dates to `2030`.
3. Go to the [Instructor Dashboard -> Membership](http://localhost:18000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-membership) and enroll `audit` as a beta tester.
4. Visit LMS as the `audit` user and check that you can submit the solution.

#### Supporting information

Private-ref: [BB-7448](https://tasks.opencraft.com/browse/BB-7448)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Agrendalath!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by Agrendalath** [comment]
@0x29a, @bradenmacdonald, this is ready for review.

@pomegranited, would you like to review this as CC?

**Comment by bradenmacdonald** [comment]
Looks good from a quick look, but I'll let others do the bulk of the review here. I'm not a CC for this repo.

**Comment by pomegranited** [comment]
@Agrendalath Sure thing, I'll review it this week.

**Review by pomegranited — APPROVED** [review]
:+1: Works great, thank you @Agrendalath !

I'll merge this tomorrow if there's no objections from anyone on [#cc-core-applications](https://openedx.slack.com/archives/C02RFB0K6GH/p1686723523086219).

- [x] I tested this on my devstack using the PR test instructions
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A -- backend changes only
- [x] Includes documentation -- +💯 for updating the erroneous links in the .github templates!
- [x] Commit structure follows [OEP-0051](https://github.com/edx/open-edx-proposals/pull/189)

**Comment by openedx-webhooks** [comment]
@Agrendalath 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/1977&entry.1484655318=https://github.com/Agrendalath&entry.752974735=2023-06-08+18:54&entry.1917517419=2023-06-14+23:37&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by pomegranited** [comment]
Merged and tagged: [v5.0.3](https://github.com/openedx/edx-ora2/tree/v5.0.3)

**Comment by Agrendalath** [comment]
Thank you, @pomegranited!
